src/services/AdvancedTradingView/adapters/advancedTradingView.test.ts

```diff
@@ -0,0 +1,136 @@
+import { ChainId, Token, UniswapV2RoutablePlatform } from '@swapr/sdk'
+
+import { combineReducers, configureStore } from '@reduxjs/toolkit'
+
+import advancedTradingView, { actions } from '../advancedTradingView.reducer'
+import { AdapterKeys, AdapterPayloadType } from '../advancedTradingView.types'
+import { adapters } from './adapters.config'
+import { AdvancedTradingViewAdapter } from './advancedTradingView.adapter'
+import { BaseAdapter } from './baseAdapter/base.adapter'
+import { PairSwaps } from './baseAdapter/base.types'
+
+jest.mock('./baseAdapter/base.adapter')
+
+const USDC_TOKEN = {
+  address: '0xA0b86991c6218b36c1d19D4a2e9Eb0cE3606eB48',
+  chainId: ChainId.MAINNET,
+  decimals: 6,
+  name: 'USDC Coin',
+  symbol: 'USDC',
+} as Token
+
+const USDT_TOKEN = {
+  chainId: ChainId.MAINNET,
+  decimals: 6,
+  name: 'Tether Coin',
+  symbol: 'USDT',
+  address: '0xdAC17F958D2ee523a2206206994597C13D831ec7',
+} as Token
+
+const adapterConstructorParams = {
+  key: AdapterKeys.SWAPR,
+  adapterSupportedChains: [ChainId.MAINNET, ChainId.ARBITRUM_ONE, ChainId.GNOSIS],
+  platform: UniswapV2RoutablePlatform.SWAPR,
+  subgraphUrls: {
+    [ChainId.MAINNET]: 'https://api.thegraph.com/subgraphs/name/dxgraphs/swapr-mainnet-v2',
+    [ChainId.ARBITRUM_ONE]: 'https://api.thegraph.com/subgraphs/name/dxgraphs/swapr-arbitrum-one-v3',
+    [ChainId.GNOSIS]: 'https://api.thegraph.com/subgraphs/name/dxgraphs/swapr-xdai-v2',
+  },
+}
+
+const store = configureStore({
+  reducer: combineReducers({ advancedTradingView }),
+})
+
+describe('AdvancedTradingView - adapter', () => {
+  const advancedTradingViewAdapter = new AdvancedTradingViewAdapter({
+    adapters,
+    chainId: ChainId.MAINNET,
+    store: store as any,
```

Comment:
if you need to do `as any` you're doing something bad :D
Remember you can make  AdvancedTradingViewAdapter take in store state as generic

```
class  AdvancedTradingViewAdapter<AppState = AppState> {
```

and pass something different in tests